be/src/exec/vectorized/hdfs_scanner.cpp

```diff
@@ -191,7 +191,8 @@ Status HdfsScanner::open_random_access_file() {
                 std::make_shared<io::CompressedInputStream>(stream, DecompressorPtr(dec.release()));
         input_stream = std::make_shared<io::CompressedSeekableInputStream>(compressed_input_stream);
     }
-    if (config::block_cache_enable) {
+
+    if (_scanner_params.use_block_cache && _compression_type == CompressionTypePB::NO_COMPRESSION) {
```

Comment:
why compressed stream  can't use block cache?

Reply:
Because there is seek implementation. For block cache, we need to align block offset.

Reply:
> Because there is seek implementation. For block cache, we need to align block offset and do seek before reading.
> 
> Although for real case, we read compressed streaming from beginning to end and there is no random IO. however for cache input stream, it has no idea of underlying input stream.

The compressed stream is only used for csv format currently?

Reply:
Yes. for compressed csv file only.